examples/notebooks/IAM_Handwriting_Recognition.ipynb

```diff
@@ -0,0 +1,5589 @@
+{
+  "nbformat": 4,
+  "nbformat_minor": 0,
+  "metadata": {
+    "colab": {
+      "name": "IAM Handwriting Recognition.ipynb",
+      "provenance": [],
+      "collapsed_sections": []
+    },
+    "kernelspec": {
+      "name": "python3",
+      "display_name": "Python 3"
+    },
+    "accelerator": "GPU"
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "A0aYnK2uLH7x"
+      },
+      "source": [
+        "In this notebook, we'll build a simple [CTC](https://www.cs.toronto.edu/~graves/icml_2006.pdf) based handwriting recognition model using [GTN](https://github.com/facebookresearch/gtn) using the [IAM Handwriting Database](https://fki.tic.heia-fr.ch/databases/iam-handwriting-database).\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "MgoDtFG8jzGt"
+      },
+      "source": [
+        "## Step 1: Setting up the environment"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Fj_VUh8djkRl"
+      },
+      "source": [
+        "\n",
+        "First, we will install gtn and import all the required modules. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "aVRKiF_7NoBX",
+        "outputId": "e54656aa-5947-480a-cec6-b048818ffed8"
+      },
+      "source": [
+        "!pip install gtn\n",
+        "import gtn \n",
+        "import os \n",
+        "import numpy\n",
+        "import shutil\n",
+        "import json \n",
+        "import random \n",
+        "from torchvision import transforms\n",
+        "import matplotlib.pyplot as plt\n",
+        "from IPython.display import display\n",
+        "import PIL \n",
+        "import numpy as np "
+      ],
+      "execution_count": 18,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "Requirement already satisfied: gtn in /usr/local/lib/python3.6/dist-packages (0.0.0)\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "sQCzZSQGMhjm"
+      },
+      "source": [
+        "Download [gtn_applications](https://github.com/facebookresearch/gtn_applications) github repo which contains scripts to download the dataset and train/evaluate models using PyTorch. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "50dH66gDOFth",
+        "outputId": "3abc5931-e5ff-4c2a-bad5-8364c193e3f9"
+      },
+      "source": [
+        "shutil.rmtree(\"gtn_applications\", ignore_errors=True)\n",
+        "!git clone https://github.com/facebookresearch/gtn_applications.git\n",
+        "assert os.path.exists(\"gtn_applications\") and os.path.isdir(\"gtn_applications\")\n",
```

Comment:
Maybe just do a check for the dir and download if it is not present, instead of removing it and downloading? This may lead to consternation if someone runs this by accident.